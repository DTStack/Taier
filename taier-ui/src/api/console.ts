/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * 系统管理
 */
import http from './http';
import req from './reqConsole';

export default {
	addTenant(params: any) {
		return http.post(req.ADD_TENANT, params);
	},
	switchTenant(params: { tenantId: number }) {
		return http.post(req.SWITCH_TENANT, params);
	},
	login(params: { username: string; password: string }) {
		return http.postAsFormData(req.LOGIN, params);
	},
	// 4.0 版本相关接口
	addCluster(params: { clusterName: string }) {
		return http.post(req.ADD_CLUSTER, params); // 新增集群
	},
	getClusterInfo(params: { clusterId: number }) {
		return http.get(req.GET_CLUSTER_INFO, params);
	},
	uploadResource(params: { fileName: any; componentType: number }) {
		return http.postAsFormData(req.UPLOAD_RESOURCE, params);
	},
	deleteComponent(params: { componentId: number }) {
		return http.post(req.DELETE_COMPONENT, params); // 删除组件
	},
	deleteCluster(params: { clusterId: number }) {
		return http.post(req.DELETE_CLUSTER, params);
	},
	testConnect(params: {
		clusterName: string;
		componentType: number;
		versionName: string;
		deployType: number | string;
	}) {
		return http.post(req.TEST_CONNECT, params);
	},
	testConnects(params: { clusterName: string }) {
		return http.post(req.TEST_CONNECTS, params);
	},
	closeKerberos(params: { componentId: number }) {
		return http.post(req.CLOSE_KERBEROS, params);
	},
	getVersionData(params?: any) {
		return http.get(req.GET_VERSION, params);
	},
	saveComponent(params: any) {
		return http.postAsFormData(req.SAVE_COMPONENT, params);
	},
	parseKerberos(params: any) {
		return http.postAsFormData(req.PARSE_KERBEROS, params);
	},
	getCompVersion(params: any) {
		return http.get(req.GET_COMPONENT_VERSION, params);
	},

	getResourceList(params: any) {
		return http.post(req.GET_RESOURCE_LIST, params);
	},
	getClusterList(params: { currentPage: number, pageSize: number, }) {
		return http.post(req.GET_CLUSTER_LIST, params);
	},
	uploadClusterResource(params: any) {
		return http.postAsFormData(req.UPLOAD_CLUSTER_RESOURCE, params);
	},
	getTenantList(params?: any) {
		return http.get(req.GET_TENANT_LIST, params);
	},
	getMetaComponent(params?: any) {
		return http.get(req.GET_META_COMPONENT, params);
	},
	testCluster(params: any) {
		return http.post(req.TEST_CLUSTER_CONNECT, params);
	},
	createCluster(params: any) {
		return http.postAsFormData(req.NEW_CLUSTER, params);
	},
	bindUserToQuere(params: any) {
		return http.post(req.BIND_USER_TO_RESOURCE, params);
	},
	updateCluster(params: any) {
		return http.postAsFormData(req.UPDATE_CLUSTER, params);
	},
	getQueueLists(params: any) {
		return http.post(req.GET_QUEUE_LISTS, params);
	},
	confirmSwitchQueue(params: any) {
		return http.post(req.CONFIRM_SWITCH_QUEUE, params);
	},
	// 4.1版本
	// 获取存储组件列表
	getComponentStore(params: any) {
		return http.post(req.GET_COMPONENTSTORE, params);
	},

	// 上传kerberos文件
	uploadKerberos(params: { kerberosFile: any; clusterId: string; componentCode: number }) {
		return http.postAsFormData(req.UPLOAD_KERBEROS, params);
	},

	// 更新krb5.conf文件
	updateKrb5Conf(params: { krb5Content: string }) {
		return http.post(req.UPDATE_KRB5CONF, params);
	},
	// 4.2版本
	getDtScriptAgentLabel(params: { agentAddress: string }) {
		return http.post(req.GET_DTSCRIPT_AGENT_LABEL, params);
	},
	getClusterComponentUser(params: {
		clusterId: string;
		needRefresh: boolean;
		componentTypeCode: number;
		agentAddress: string;
	}) {
		return http.post(req.GET_CLUSTER_COMPONENT_USER, params);
	},
	addOrUpdateComponentUser(params: { componentUserList: any[] }) {
		return http.post(req.ADD_OR_UPDATE_COMPONENT_USER, params);
	},
	// 任务管理模块
	// 概览-获取集群
	getClusterDetail(params: any) {
		return http.post(req.GET_CLUSTER_DETAIL, params);
	},
	// 概览-集群下拉列表
	getClusterSelect(params: any) {
		return http.post(req.GET_CLUSTER_SELECT, params);
	},
	// 获取节点下拉
	getNodeAddressSelect(params?: any) {
		return http.post(req.GET_NODEADDRESS_SELECT, params);
	},
	// 根据节点搜索

	// 明细-根据任务名搜索任务
	searchTaskList(params: any) {
		return http.post(req.SEARCH_TASKNAME_LIST, params);
	},
	// 明细-模糊查询任务名
	searchTaskFuzzy(params: any) {
		return http.post(req.SEARCH_TASKNAME_FUZZY, params);
	},
	// 明细-杀死选中或者杀死全部任务
	killTasks(params: {
		jobResource: string;
		nodeAddress: string;
		stage: number;
		jobIdList?: any[];
	}) {
		return http.post(req.KILL_TASKS, params);
	},

	killAllTask(params: { jobResource: string; nodeAddress: string }) {
		return http.post(req.KILL_ALL_TASK, params);
	},

	stickJob(params: { jobId: string; jobResource: string }) {
		return http.post(req.JOB_STICK, params);
	},
	// 引擎列表
	getEngineList(params?: any) {
		return http.post(req.GET_ENGINE_LIST, params);
	},
	// group列表
	getGroupList(params: any) {
		return http.post(req.GET_GROUP_LIST, params);
	},
	// 查看明细 和搜索条件
	getViewDetail(params: {
		stage: number;
		jobResource: string;
		nodeAddress?: string;
		currentPage: number;
		pageSize: number;
	}) {
		return http.post(req.GET_VIEW_DETAIL, params);
	},
	// 顺序调整调整优先级
	changeJobPriority(params: any) {
		return http.post(req.CHANGE_JOB_PRIORITY, params);
	},
	// 查看剩余资源
	getClusterResources(params: any) {
		return http.post(req.GET_CLUSTER_RESOURCES, params);
	},
	getLoadTemplate(params: any) {
		return http.post(req.GET_LOADTEMPLATE, params);
	},
	uploadKerberosFile(params: any) {
		return http.postAsFormData(req.UPLOAD_KERBEROSFILE, params);
	},
	getKerberosFile(params: any) {
		return http.post(req.GET_KERBEROSFILE, params);
	},
	testComponent(params: any) {
		return http.post(req.TEST_COMPONENT_CONNECT, params);
	},
	testComponentKerberos(params: any) {
		return http.postAsFormData(req.TEST_COMPONENT_CONNECT_KERBEROS, params);
	},
	addComponent(params: any) {
		return http.post(req.ADD_COMPONENT, params);
	},
	saveComponentWithKerberos(params: any) {
		return http.postAsFormData(req.SAVE_COMPONENT_KERBEROS, params);
	},
	deleteKerberos(params: any) {
		return http.post(req.DELETE_KERBEROS, params); // 删除Haddop Kerberos认证文件
	},
	addEngine(params: any) {
		return http.post(req.ADD_ENGINE, params);
	},
	addEngines(params: any) {
		return http.post(req.ADD_ENGINS, params);
	},
	updateClusterVersion(params: {
		clusterId: number;
		hadoopVersion: string;
		syncType: number /* 同步元数据组件类型 */;
	}) {
		return http.post(req.UPDATE_CLUSTER_VERSION, params);
	},
	// 资源管理
	getAllCluster(params?: any) {
		return http.get(req.GET_ALL_CLUSTER, params); // 返回数据包含集群下的engine，以及队列
	},
	getEnginesByCluster(params?: any) {
		return http.get(req.GET_ENGINES_BY_CLUSTER, params);
	},
	searchTenant(params: any) {
		return http.post(req.SEARCH_TENANT, params);
	},
	getTaskResourceTemplate(params?: any) {
		return http.post(req.TASK_RESOURCE, params);
	},

	getQueue(params: any) {
		return http.post(req.GET_QUEUE, params);
	},
	bindTenant(params: any) {
		return http.post(req.BIND_TENANT, params);
	},
	switchQueue(params: any) {
		return http.post(req.SWITCH_QUEUE, params);
	},
	bindNamespace(params: any) {
		return http.post(req.BIND_NAME_SPACE, params);
	},
	queryTaskResourceLimits(params: any) {
		return http.post(req.GET_TASKLIMITS, params);
	},

	/**
	 * 告警通道
	 */
	addOrUpdateAlarmRule(params: {
		alarmType: number;
		alertGateCode: number;
		alertGateSource: string;
		alertGateName: string;
		alertGateJson: string;
		alertTemplate: string;
		receiveMethod?: string;
		defaultChannel?: boolean;
		file?: any;
	}) {
		return http.postAsFormData(req.ADD_OR_UPDATE_ALARM, params);
	},
	getAlarmRuleList(params: {
		currentPage: number;
		pageSize: number;
		alertGateType?: any[];
		clusterId?: string;
	}) {
		return http.post(req.GET_ALARM_RULE_LIST, params);
	},
	deleteAlarmRule(params: { id: number }) {
		return http.post(req.DEL_ALARM_RULE_LIST, params);
	},
	getByAlertId(params: { id: number }) {
		return http.post(req.GET_ALARM_BY_ID, params);
	},
	setDefaultAlert(params: { alertId: number; alertGateType: number }) {
		return http.post(req.SET_DEFAULT_ALERT, params);
	},
	testAlert(params: any) {
		return http.postAsFormData(req.TEST_ALERT, params);
	},
	refreshQueue(params: { clusterName: string }) {
		return http.post(req.REFRESH_QUEUE, params);
	},
	getAlarmConfig() {
		return http.post(req.GET_ALARM_CONFIG);
	},
	updateAlarmConfig(params: { componentConfig: string; componentTemplate: string }) {
		return http.post(req.UPDATE_ALARM_CONFIG, params);
	},
	testAlarmConfig() {
		return http.post(req.TEST_ALARM_CONFIG);
	},
};
