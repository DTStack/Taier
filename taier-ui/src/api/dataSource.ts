/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { queryParse } from '@/utils';
import Swagger from './dataSources-swagger';
import http from './http';

type APIProps = Record<
	keyof typeof Swagger,
	(params: any, config?: any) => Promise<any>
>;

function mapUrlObjToFuncObj(urlObj: typeof Swagger) {
	const keys = Object.keys(urlObj) as (keyof typeof Swagger)[];
	return keys.reduce((pre, cur) => {
		const item = urlObj[cur];
		pre[cur] = async function (params, config: any = {}) {
			const queryParams: any = queryParse(window.location.hash);
			// 所有的请求接口统一带上header
			if (queryParams.datasource_id) {
				config.headers = new Headers({
					dt_datasource_id: queryParams.datasource_id,
				});
			}
			return await http[item.method as 'get' | 'post' | 'postForm'](
				item.url,
				params,
				config,
			);
		};
		return pre;
	}, {} as APIProps);
}
// function mapUrlObjToStrObj(urlObj) {
//   const Url = {};
//   keys(urlObj).forEach((key) => {
//     const item = urlObj[key];
//     Url[key] = item.url;
//   });
//   return Url;
// }

export const API = mapUrlObjToFuncObj(Object.assign({}, Swagger));
// export const URL = mapUrlObjToStrObj(Object.assign({}, Swagger, Restful));
