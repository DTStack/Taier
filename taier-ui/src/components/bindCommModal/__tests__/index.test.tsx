import api from '@/api';
import { fireConfirmOnModal, selectItem, toggleOpen } from '@/tests/utils';
import { cleanup, render, waitFor } from '@testing-library/react';
import BindCommModal from '..';
import '@testing-library/jest-dom';

jest.mock('@/api');

describe('Test BindCommModal Component', () => {
    beforeEach(() => {
        cleanup();
        document.body.innerHTML = '';
        jest.useFakeTimers();
        (api.getTenantList as jest.Mock).mockReset().mockResolvedValue({
            code: 1,
            data: [
                {
                    tenantId: 1,
                    tenantName: 'test',
                },
            ],
        });

        (api.getEnginesByCluster as jest.Mock).mockReset().mockResolvedValue({
            code: 1,
            data: {
                engines: [
                    {
                        id: 0,
                        gmtCreate: null,
                        gmtModified: null,
                        isDeleted: 0,
                        components: null,
                        clusterId: -1,
                        engineName: 'Hadoop',
                        engineType: 1,
                    },
                ],
            },
        });

        (api.getClusterResources as jest.Mock).mockReset().mockResolvedValue({
            code: 1,
            data: {
                queues: [
                    {
                        queueName: 'queueName',
                    },
                ],
            },
        });
    });

    it('Should match snapshot', () => {
        const { asFragment } = render(
            <BindCommModal
                title="绑定新租户"
                visible
                clusterList={[{ clusterId: 1, clusterName: 'test' }] as any}
                isBindTenant
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('Should log errors', async () => {
        const { getByTestId, getByText, container, asFragment } = render(
            <BindCommModal
                title="绑定新租户"
                visible
                clusterList={[{ clusterId: 1, clusterName: 'test' }] as any}
                isBindTenant
            />
        );

        fireConfirmOnModal(getByTestId);

        await waitFor(() => {
            expect(getByText('租户不可为空！')).toBeInTheDocument();
            expect(getByText('集群不可为空！')).toBeInTheDocument();
        });

        toggleOpen(container.querySelectorAll<HTMLElement>('.ant-form-item')[1]);
        selectItem();

        await waitFor(() => {
            expect(api.getEnginesByCluster).toBeCalled();
            expect(api.getClusterResources).toBeCalled();
        });

        expect(asFragment()).toMatchSnapshot('Show queue field');
    });

    it('Should confirm successfully', async () => {
        const fn = jest.fn();
        const { getByTestId, container } = render(
            <BindCommModal
                title="绑定新租户"
                visible
                clusterList={[{ clusterId: 1, clusterName: 'test' }] as any}
                isBindTenant
                onOk={fn}
            />
        );

        await waitFor(() => {
            expect(api.getTenantList).toBeCalled();
        });

        toggleOpen(container.querySelectorAll<HTMLElement>('.ant-form-item')[0]);
        selectItem();

        toggleOpen(container.querySelectorAll<HTMLElement>('.ant-form-item')[1]);
        selectItem(0, 1);

        await waitFor(() => {
            expect(api.getEnginesByCluster).toBeCalled();
            expect(api.getClusterResources).toBeCalled();
        });

        toggleOpen(container.querySelectorAll<HTMLElement>('.ant-form-item')[2]);
        selectItem(0, 2);

        fireConfirmOnModal(getByTestId);

        await waitFor(() => expect(fn).toBeCalledWith({ tenantId: 1, clusterId: 1, queueName: 'queueName' }));
    });
});
