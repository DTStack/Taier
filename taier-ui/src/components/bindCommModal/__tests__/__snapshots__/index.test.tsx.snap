// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BindCommModal Component Should log errors: Show queue field 1`] = `
<DocumentFragment>
  <p
    data-testid="antd-mock-Modal-title"
  >
    绑定新租户
  </p>
  <div
    class="info-title"
  >
    <span
      aria-label="info-circle"
      class="anticon anticon-info-circle"
      role="img"
      style="color: rgb(36, 145, 247);"
    >
      <svg
        aria-hidden="true"
        data-icon="info-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
        <path
          d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
        />
      </svg>
    </span>
    <span
      class="info-text"
    >
      将租户绑定到集群，可使用集群内的每种计算引擎，绑定后，不能切换其他集群。
    </span>
  </div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-form-item ant-form-item-with-help ant-form-item-has-error"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="tenantId"
            title="租户"
          >
            租户
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                aria-describedby="tenantId_help"
                aria-invalid="true"
                aria-required="true"
                class="ant-select ant-select-in-form-item ant-select-status-error ant-select-single ant-select-allow-clear ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="tenantId_list_0"
                      aria-autocomplete="list"
                      aria-controls="tenantId_list"
                      aria-describedby="tenantId_help"
                      aria-haspopup="listbox"
                      aria-invalid="true"
                      aria-owns="tenantId_list"
                      aria-required="true"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="tenantId"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  >
                    请搜索要绑定的租户
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            style="display: flex; flex-wrap: nowrap;"
          >
            <div
              class="ant-form-item-explain ant-form-item-explain-connected"
              id="tenantId_help"
              role="alert"
            >
              <div
                class="ant-form-item-explain-error"
              >
                租户不可为空！
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item ant-form-item-has-success"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="clusterId"
            title="集群"
          >
            集群
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                aria-required="true"
                class="ant-select ant-select-in-form-item ant-select-status-success ant-select-single ant-select-allow-clear ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="clusterId_list_0"
                      aria-autocomplete="list"
                      aria-controls="clusterId_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-owns="clusterId_list"
                      aria-required="true"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="clusterId"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                    title="test"
                  >
                    test
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  aria-hidden="true"
                  class="ant-select-clear"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-form-item"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
            >
              <label
                class="ant-form-item-required"
                for="queueName"
                title="队列"
              >
                队列
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    aria-required="true"
                    class="ant-select ant-select-in-form-item ant-select-single ant-select-allow-clear ant-select-show-arrow"
                  >
                    <div
                      class="ant-select-selector"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="queueName_list_0"
                          aria-autocomplete="list"
                          aria-controls="queueName_list"
                          aria-haspopup="listbox"
                          aria-owns="queueName_list"
                          aria-required="true"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="queueName"
                          readonly=""
                          role="combobox"
                          style="opacity: 0;"
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-placeholder"
                      >
                        请选择队列
                      </span>
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        class="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <button
    data-testid="antd-mock-Modal-confirm"
  >
    confirm
  </button>
</DocumentFragment>
`;

exports[`Test BindCommModal Component Should match snapshot 1`] = `
<DocumentFragment>
  <p
    data-testid="antd-mock-Modal-title"
  >
    绑定新租户
  </p>
  <div
    class="info-title"
  >
    <span
      aria-label="info-circle"
      class="anticon anticon-info-circle"
      role="img"
      style="color: rgb(36, 145, 247);"
    >
      <svg
        aria-hidden="true"
        data-icon="info-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
        />
        <path
          d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
        />
      </svg>
    </span>
    <span
      class="info-text"
    >
      将租户绑定到集群，可使用集群内的每种计算引擎，绑定后，不能切换其他集群。
    </span>
  </div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="tenantId"
            title="租户"
          >
            租户
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                aria-required="true"
                class="ant-select ant-select-in-form-item ant-select-single ant-select-allow-clear ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="tenantId_list_0"
                      aria-autocomplete="list"
                      aria-controls="tenantId_list"
                      aria-haspopup="listbox"
                      aria-owns="tenantId_list"
                      aria-required="true"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="tenantId"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  >
                    请搜索要绑定的租户
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="clusterId"
            title="集群"
          >
            集群
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                aria-required="true"
                class="ant-select ant-select-in-form-item ant-select-single ant-select-allow-clear ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="clusterId_list_0"
                      aria-autocomplete="list"
                      aria-controls="clusterId_list"
                      aria-haspopup="listbox"
                      aria-owns="clusterId_list"
                      aria-required="true"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="clusterId"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  >
                    请选择集群
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <button
    data-testid="antd-mock-Modal-confirm"
  >
    confirm
  </button>
</DocumentFragment>
`;
